### PR TITLE
[stable/kibana] legacy-ingress-api-version

### PR DESCRIPTION
**Pull request by japzio**
Signed-off-by: Jasper Culong <jculongit10@yahoo.com>

#### Is this a new chart

NO

#### What this PR does / why we need it:

To support legacy ingress api version extensions/v1beta1 aside from the networking.k8s.io/v1beta1

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
Thanks for your contribution but i dont think the api version should be a configurable value.
Instead you could use something like this, regarding to the K8s version:

```
{{- if semverCompare "> 1.19" .Capabilities.KubeVersion.GitVersion }}
apiVersion: networking.k8s.io/v1beta1
{{ else }}
apiVersion: extensions/v1beta1
{{- end }}
```

**Comment by japzio** [comment]
> Thanks for your contribution but i dont think the api version should be a configurable value.
> Instead you could use something like this, regarding to the K8s version:
> 
> ```
> {{- if semverCompare "> 1.19" .Capabilities.KubeVersion.GitVersion }}
> apiVersion: networking.k8s.io/v1beta1
> {{ else }}
> apiVersion: extensions/v1beta1
> {{- end }}
> ```

Hi @monotek, just applied your suggestion.

**Review by monotek — CHANGES_REQUESTED** [review]
edit...

File: stable/kibana/values.yaml [hunk 1]
```diff
@@ -78,6 +78,7 @@ service:
 
 ingress:
   enabled: false
+  apiVersion: networking.k8s.io/v1beta1
```
Comment (monotek) [inline stable/kibana/values.yaml]: please remove line

**Review by monotek — CHANGES_REQUESTED** [review]

File: stable/kibana/README.md [hunk 2]
```diff
@@ -59,6 +59,7 @@ The following table lists the configurable parameters of the kibana chart and th
 | `image.pullSecrets`                        | Specify image pull secrets                                             | `nil`                                 |
 | `commandline.args`                         | add additional commandline args                                        | `nil`                                 |
 | `ingress.enabled`                          | Enables Ingress                                                        | `false`                               |
+| `ingress.apiVersion`                       | Ingress API Version                                                    | `networking.k8s.io/v1beta1`                               |
```
Comment (monotek) [inline stable/kibana/README.md]: please remove line

File: stable/kibana/templates/ingress.yaml [hunk 3]
```diff
@@ -1,7 +1,11 @@
 {{- if .Values.ingress.enabled -}}
 {{- $serviceName := include "kibana.fullname" . -}}
 {{- $servicePort := .Values.service.externalPort -}}
+{{- if semverCompare "> 1.19" .Capabilities.KubeVersion.GitVersion }}
```
Comment (monotek) [inline stable/kibana/templates/ingress.yaml]: Sorry, for using an example version here.

New api will work form Kubernetes 1.14: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.14.md#deprecations

So it should be ">= 1.14"

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by japzio** [comment]
Oh yeah2x i came across that doc.

**Review by monotek — CHANGES_REQUESTED** [review]
Looks good to me.

But please sign your commits to get the DCO check passed.
See: https://github.com/helm/charts/pull/18974/checks?check_run_id=309996144

**Comment by japzio** [comment]
Hi @monotek Check resolved thanks. :-)

**Review by monotek — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**